cocos/2d/CCTMXXMLParser.cpp

```diff
@@ -746,7 +746,7 @@ void TMXMapInfo::endElement(void *ctx, const char *name)
 
             uint32_t* bufferPtr = reinterpret_cast<uint32_t*>(buffer);
             for(auto gidToken : gidTokens) {
-                auto tileGid = (uint32_t)strtol(gidToken.c_str(), nullptr, 10);
```

Comment:
if gid is 1, after using horizontal flip, it will be " 1 & kTMXTileHorizontalFlag" = 0x80000001